Bring up board for ESP32-S3-Matrix

xingrz opened the pull request:
ESP32-S3-Matrix is a tiny board from Waveshare, shipped with an ESP32-S3FH4R2, a QMI8658C IMU and an 8x8 WS2812 RGB matrix.

* https://www.waveshare.com/wiki/ESP32-S3-Matrix

---

#### `samples/drivers/led/led_strip`

https://github.com/user-attachments/assets/a24bac13-b00c-4244-baf6-a920b74715a4

#### `samples/drivers/display`

(with `"led-strip-matrix"`)

![958211738060843_ pic_hd](https://github.com/user-attachments/assets/a0a7a6bd-f641-49e6-886d-b0433a28ebaf)

xingrz committed:
boards: waveshare: esp32s3_matrix: Bring up board for ESP32-S3-Matrix

ESP32-S3-Matrix is a tiny board from Waveshare, shipped with an
ESP32-S3FH4R2, a QMI8658C IMU and an 8x8 WS2812 RGB matrix.

Signed-off-by: Chen Xingyu <hi@xingrz.me>

nordicjm commented on boards/waveshare/esp32s3_matrix/Kconfig:
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2025 Chen Xingyu <hi@xingrz.me>
+# SPDX-License-Identifier: Apache-2.0
+
+config HEAP_MEM_POOL_ADD_SIZE_BOARD
+	int
+	default 4096 if BOARD_ESP32S3_MATRIX_ESP32S3_PROCPU
+	default 256 if BOARD_ESP32S3_MATRIX_ESP32S3_APPCPU
```
think this has been replaced? @marekmatej

xingrz replied:
It's required by `modules/hal/espressif/zephyr/port/heap/heap_caps_zephyr.c`

nordicjm replied:
ping @marekmatej

marekmatej replied:
@nordicjm Actually, this heap request comes from the boards, so I don't think its wrong.

nordicjm commented on boards/waveshare/esp32s3_matrix/doc/index.rst:
```diff
@@ -0,0 +1,111 @@
+Supported Features
+==================
+
+Current Zephyr's ESP32-S3-Matrix board supports the following features:
```
```suggestion
Currently Zephyr's ``esp32s3_matrix`` board supports the following features:
```

nordicjm commented on boards/waveshare/esp32s3_matrix/esp32s3_matrix_esp32s3_procpu.dts:
```diff
@@ -0,0 +1,103 @@
+
+	buttons {
+		compatible = "gpio-keys";
+		button0: button_0 {
```
inconsistent spacing in file

nordicjm commented on boards/waveshare/esp32s3_matrix/esp32s3_matrix_esp32s3_procpu.dts:
```diff
@@ -0,0 +1,103 @@
+		compatible = "worldsemi,ws2812-spi";
+		reg = <0>;
+		spi-max-frequency = <WS2812C_SPI_FREQ>;
+
```
```suggestion
```

nordicjm commented on boards/waveshare/esp32s3_matrix/esp32s3_matrix_esp32s3_procpu.dts:
```diff
@@ -0,0 +1,103 @@
+	#size-cells = <0>;
+	pinctrl-0 = <&spim2_ws2812_led>;
+	pinctrl-names = "default";
+
```
```suggestion
```

nordicjm commented on boards/waveshare/esp32s3_matrix/esp32s3_matrix_esp32s3_procpu_defconfig:
```diff
@@ -0,0 +1,12 @@
+# Copyright (c) 2025 Chen Xingyu <hi@xingrz.me>
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_MAIN_STACK_SIZE=2048
+CONFIG_GPIO=y
+CONFIG_CONSOLE=y
+CONFIG_SERIAL=y
+CONFIG_UART_CONSOLE=y
+CONFIG_CLOCK_CONTROL=y
+CONFIG_INPUT=y
+CONFIG_LED_STRIP=y
```
no main stack size, no input, no led strip

nordicjm replied:
Since it's a driver within a driver of another class, an overlay for the sample is fine

xingrz committed:
drivers: display: Select LED_STRIP when LED_STRIP_MATRIX is enabled

The LED_STRIP driver is not enabled by default on all boards. As a result,
building the LED_STRIP_MATRIX display driver fails if LED_STRIP is not
manually enabled.

To address this, this commit changes the dependency from `depends on` to
`select`, ensuring that the required driver is automatically enabled and
build failures are avoided.

Signed-off-by: Chen Xingyu <hi@xingrz.me>

nordicjm reviewed (commented):
Changes OK pending comment

danieldegrasse reviewed (dismissed):
Display changes look fine- will defer to nordicjm for review on the board changes though.

marekmatej commented on boards/waveshare/esp32s3_matrix/Kconfig.defconfig:
```diff
@@ -0,0 +1,10 @@
+
+if BOARD_ESP32S3_MATRIX_ESP32S3_PROCPU
+
+config KERNEL_MEM_POOL
```
This config is not necessary - enabled by default.

marekmatej commented on boards/waveshare/esp32s3_matrix/doc/index.rst:
```diff
@@ -0,0 +1,111 @@
+
+- Dual core 32-bit Xtensa Microprocessor (Tensilica LX7), running up to 240MHz
+- Additional vector instructions support for AI acceleration
+- 2MB of SRAM
```
this board doesn't have 2MB of SRAM.

marekmatej commented on boards/waveshare/esp32s3_matrix/doc/index.rst:
```diff
@@ -0,0 +1,111 @@
+Building & Flashing
+*******************
+
+Simple boot
```
For some reason your .rst is missing MCUboot and Sysbuild chapters.

marekmatej reviewed (changes requested):
Overall LGTM! Some minor changes needs to be done though.

kartben commented on boards/waveshare/esp32s3_matrix/doc/index.rst:
```diff
@@ -0,0 +1,111 @@
+Supported Features
+==================
+
+Currently Zephyr's ``esp32s3_matrix`` board supports the following features:
+
++------------+------------+-------------------------------------+
+| Interface  | Controller | Driver/Component                    |
++============+============+=====================================+
+| UART       | on-chip    | serial port                         |
++------------+------------+-------------------------------------+
+| GPIO       | on-chip    | gpio                                |
++------------+------------+-------------------------------------+
+| PINMUX     | on-chip    | pinmux                              |
++------------+------------+-------------------------------------+
+| SPI Master | on-chip    | spi                                 |
++------------+------------+-------------------------------------+
+| Watchdog   | on-chip    | watchdog                            |
++------------+------------+-------------------------------------+
+| TRNG       | on-chip    | entropy                             |
++------------+------------+-------------------------------------+
```
```suggestion
.. zephyr:board-supported-hw::
```

kartben commented on boards/waveshare/esp32s3_matrix/doc/index.rst:
```diff
@@ -0,0 +1,111 @@
+
+Building & Flashing
+*******************
+
```
```suggestion

.. zephyr:board-supported-runners::

```

kartben reviewed (changes requested):
Some minor comments re: documentation page. @xingrz will you be able to come back to this PR?

kartben commented:
@xingrz just checking if you still have this PR on your radar?
Thanks!

kartben reviewed (approved):
Nice!
